storage/txcache/txCache.go

```diff
@@ -0,0 +1,99 @@
+package txcache
+
+import (
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+)
+
+// TxCache is
+type TxCache struct {
+	txListBySender   TxListBySenderMap
+	txByHash         TxByHashMap
+	EvictionStrategy *EvictionStrategy
+}
+
+// NewTxCache creates a new transaction cache
+// "size" dictates the maximum number of transactions to hold in this cache at a given time
+// "shardsHint" is used to configure the internal concurrent maps on which the implementation relies
+func NewTxCache(size int, shardsHint int) *TxCache {
```

Comment:
this should get txByHash and txListBySender as an input parameter. And use a factory to set up all the txCache.

Reply:
`TxCache` owns `txByHash` and `txListBySender`. It is composition, not aggregation. Therefore, `TxCache` wants to explicitly instantiate and manage the two owned objects.